R/object-builder.R

```diff
@@ -27,9 +27,9 @@ build_dfm <- function(x, features,
                       docvars = data.frame(), meta = list(), 
                       class = "dfm", ...) {
     result <- new(class,
-        as(x, "dgCMatrix"),
-        docvars = docvars,
-        meta = make_meta("dfm", inherit = meta, ...)
+                  as(as(as(x, "CsparseMatrix"), "generalMatrix"), "dMatrix"),
```

Comment:
DFM is dgC or dgT when created internally. It is necessary to apply `as(as(as(x, "CsparseMatrix"), "generalMatrix"), "dMatrix")` only when `x` is unknown format. We probably need triple coercion for `as.dfm()` where `x` can be any matrix.

Reply:
I left it here since it achieves the purpose of the original code, which was to ensure that build_dfm returns a dgCMatrix. I think you were suggesting a broader change that this glimpse at the code revealed. First things first - getting Matrix working. Did I misunderstand?

Reply:
Misunderstanding. I am suggesting to make simplify the code to make it work with Matrix 1.4-2: if `x` is dgT or dgC, you need only `as(x, "CsparseMatrix")`. `as(as(x, "generalMatrix"), "dMatrix")` is redundant.